server/public/model/integration_action.go

```diff
@@ -335,6 +347,130 @@ func (r *OpenDialogRequest) DecodeAndVerifyTriggerId(s *ecdsa.PrivateKey, timeou
 	return DecodeAndVerifyTriggerId(r.TriggerId, s, timeout)
 }
 
+func (r *OpenDialogRequest) IsValid() error {
+	var multiErr *multierror.Error
+	if r.URL == "" || !IsValidHTTPURL(r.URL) {
+		multiErr = multierror.Append(multiErr, errors.New("invalid URL"))
+	}
+
+	if r.TriggerId == "" {
+		multiErr = multierror.Append(multiErr, errors.New("empty trigger id"))
+	}
+
+	err := r.Dialog.IsValid()
+	if err != nil {
+		multiErr = multierror.Append(multiErr, err)
+	}
+	return multiErr.ErrorOrNil()
+}
+
+func (d *Dialog) IsValid() error {
+	var multiErr *multierror.Error
+
+	if d.Title == "" || len(d.Title) > DialogTitleMaxLength {
+		multiErr = multierror.Append(multiErr, errors.New("invalid dialog title"))
+	}
+
+	if d.IconURL != "" && !IsValidHTTPURL(d.IconURL) {
+		multiErr = multierror.Append(multiErr, errors.New("invalid icon url"))
+	}
+	if len(d.Elements) > DialogMaxElements {
+		multiErr = multierror.Append(multiErr, errors.Errorf("maximum number of dialog elements is %d", DialogMaxElements))
+	}
```

Comment:
That is actually no longer true https://github.com/mattermost/mattermost-developer-documentation/pull/1341